sycl/source/detail/kernel_info.hpp

```diff
@@ -178,6 +205,13 @@ inline uint32_t get_kernel_device_specific_info_host<
                              PI_ERROR_INVALID_KERNEL);
 }
 
+template <>
+inline uint32_t get_kernel_device_specific_info_host<
+    info::kernel_device_specific::max_sub_group_size>(const sycl::device &) {
+  throw invalid_object_error("This instance of kernel is a host instance",
+                             PI_ERROR_INVALID_KERNEL);
+}
+
```

Comment:
Should it be an assert instead? Host device isn't available to the user, so it should totally under our control if we can ever have it here.

Reply:
We could, yes, though technically that's not the case yet.

Reply:
What do you mean?

Reply:
The host device is not yet removed. Technically speaking it could still be picked in the time between this being merged and the patch removing it is merged.